libraries/AP_NavEKF3/AP_NavEKF3_MagFusion.cpp

```diff
@@ -127,6 +127,9 @@ void NavEKF3_core::controlMagYawReset()
                 gcs().send_text(MAV_SEVERITY_WARNING, "EKF3 IMU%u ground mag anomaly, yaw re-aligned",(unsigned)imu_index);
             }
 
+            // prevent reset of variances in ConstrainVariances()
+            inhibitMagStates = false;
```

Comment:
This will result in all subsequent fusion operations on that frame processing the additional states for the earth/body mag field. However given that the off-diagonals in the covariance matrix will have been set to zero in the calcQuatAndFieldStates() function, there should be no resultant modification of field states for any subsequent fusion operations and the corret value for inhibitMagStates will be set at the start of the next frame.

This means that the only negative i can see for this change is that i the case where we are doing a yaw reset whilst intending to remain in heading fusion mode, that there will be a small computational penalty for any subsequent fusion operations on that frame only.

Reply:
ok, I can see the small cost. I think it is acceptable